app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,12 @@
+class RecipesController < ApplicationController
+  def show
+    @recipe = Recipe.find(params[:id])
+    @ingredients = @recipe.ingredients
+    @total_cost = @ingredients.sum(:cost)
```

Comment:
While this AR is correct, it's living inside this controller which is an MVC infraction. AR that queries the database for a calculation or a dynamic data set should be in the model, and it should be tested. 

I would call this method something like `total_cost`, and put essentially the same AR you wrote in that method, and either call it here in the controller or directly in the view, either are fine.